collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1806,7 +1801,6 @@ public void onClick(DialogInterface dialog, int i) {
                 switch (i) {
                     case BUTTON_POSITIVE: // yes
                         clearAnswer(qw);
-                        formEntryViewModel.updateAnswersForScreen(getAnswers(), false);
```

Comment:
This indeed seems to be redundant but we would need to ask the QA team to play with clearing answers.